packages/frontend-2/plugins/dayjs.ts

```diff
@@ -7,4 +7,44 @@ export default defineNuxtPlugin(() => {
   dayjs.extend(relativeTime)
   dayjs.extend(localizedFormat)
   dayjs.extend(duration)
+
+  const customRelativeTime = (date: string): string => {
+    const pastDate = dayjs(date)
+    const now = dayjs()
+    const diffInMinutes = now.diff(date, 'minute')
+    const diffInHours = now.diff(date, 'hour')
+    const diffInDays = now.diff(date, 'day')
+
+    if (diffInDays > 14) {
+      return pastDate.year() === now.year()
+        ? pastDate.format('MMM D')
+        : pastDate.format('MMM D, YYYY')
+    } else if (diffInDays >= 1) {
+      return diffInDays === 1 ? '1 day ago' : `${diffInDays} days ago`
+    } else if (diffInHours <= 23 && diffInHours >= 1) {
+      return diffInHours === 1 ? '1 hour ago' : `${diffInHours} hours ago`
+    } else if (diffInMinutes <= 59 && diffInMinutes >= 1) {
+      return diffInMinutes === 1 ? '1 minute ago' : `${diffInMinutes} minutes ago`
+    }
+
+    return 'Just now'
+  }
+
+  const isClockUnit = (date: string) => {
+    const unit = customRelativeTime(date)
+    return unit.includes('second') || unit.includes('minute') || unit.includes('hour')
+  }
+
+  const getFullDate = (input: string) => dayjs(input).format('MMM D, YYYY, H:mm')
+  const getTrunicatedRelativeDate = (input: string) => customRelativeTime(input)
+  const getTrunicatedRelativeDateWithPrefix = (input: string) =>
```

Comment:
1. Trunicated - misspelling
2. `input: string` should be optional, because often you want dayjs() to work with the current date. Also, it can be way more than a string - the dayjs() factory also accepts numbers, other Dates, other Dayjs instances. Make the input type the same one that the factory accepts: `dayjs.ConfigType`
3. `getTruncatedRelativeDateWithPrefix` doesn't have to be its own function, you can just add that as an optional param in the other function - `getTruncatedRelativeDate(input: string, options?: Partial<{prefix: boolean}>)`
4. These names are too long IMO. I'd just have `fullDate()` and `relativeDate()`

Reply:
@fabis94 Updated :)